### PR TITLE
Add .gitignore

### PR DESCRIPTION
**Pull request by KarolisPetrovskis**
Added a .gitignore file and removed files that should not have been tracked.

### CODE REVIEW

**Review by Vanielius1013 — APPROVED** [review]
Seems legit

**Review by Vanielius1013 — APPROVED** [review]
nice tidying up.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+
+# Editor directories and files
+.vscode/*
+!.vscode/extensions.json
+.idea
+.DS_Store
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
```
Comment (KarolisPetrovskis) [inline .gitignore:508]: Probably should keep .sln files for convenience

**Comment by KarolisPetrovskis** [comment]
Will close this and open new PR, seems like the easiest way to restore .sln file deleted here

**Pull request closed without merging**